Sources/SMTPKitten/Connection/SMTPClient.swift

```diff
@@ -0,0 +1,157 @@
+import NIOCore
+import NIOPosix
+import NIOExtras
+import NIOSSL
+
+public actor SMTPClient {
+    let channel: NIOAsyncChannel<SMTPReplyLine, ByteBuffer>
+    fileprivate let requests: AsyncStream<SMTPRequest>
+    fileprivate let requestWriter: AsyncStream<SMTPRequest>.Continuation
+    fileprivate var error: Error?
+    fileprivate var _handshake: SMTPHandshake?
+    internal var handshake: SMTPHandshake {
+        guard let _handshake else {
+            preconditionFailure("SMTPClient didn't set the SMTPHandshake after getting it")
+        }
+
+        return _handshake
+    }
+
+    fileprivate init(channel: NIOAsyncChannel<SMTPReplyLine, ByteBuffer>) {
+        self.channel = channel
+        (requests, requestWriter) = AsyncStream.makeStream(of: SMTPRequest.self, bufferingPolicy: .unbounded)
```

Comment:
This obviously doesn't provide back pressure, but for a client I guess this is of less concern

Reply:
Yeah I think it's of minimal concern here.